File: BinaryTree/python/treeIncludes.py
from collections import deque
from typing import Optional

from node import Node


def tree_includes(root: Optional[Node], target: int) -> bool:
    if root is None:
        return False

    queue = deque([root])
    while queue:
        current = queue.popleft()
        if current.val == target:
            return True
        if current.left:
            queue.append(current.left)
        if current.right:
            queue.append(current.right)
    return False

File: BinaryTree/go/treeIncludes.go
package binarytree

// TreeIncludes reports whether target exists in the tree.
func TreeIncludes(root *TreeNode, target int) bool {
	if root == nil {
		return false
	}

	queue := []*TreeNode{root}
	for len(queue) > 0 {
		node := queue[0]
		queue = queue[1:]

		if node.Val == target {
			return true
		}
		if node.Left != nil {
			queue = append(queue, node.Left)
		}
		if node.Right != nil {
			queue = append(queue, node.Right)
		}
	}
	return false
}
